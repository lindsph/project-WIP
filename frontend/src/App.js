import React, {useState, useEffect} from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link,
  NavLink
} from "react-router-dom";
import axios from 'axios';
import Landing from './Landing';
import Dashboard from './Dashboard';
import Form from './Form';
import ErrorPage from './ErrorPage';

const App = () => {
  const [isLoading, setIsLoading] = useState(false);

  const handleLogin = (email, password) => {
    // temporary measure, add to db and add as context?
    setIsLoading(true);
    const response = axios.get('http://localhost:5000/users')
      .then(response => {
        const users = response.data.users;
        console.log(users);
      })
      .catch(error => {
        console.log(error)
      })
  }

  return (
    <Router>
      <>
      <header>
        <nav>
          <NavLink to="/sign_in">Sign in</NavLink>
        </nav>
      </header>
      <Switch>
        <Route exact path="/" component={Landing}></Route>
        <Route path="/sign_in">
          <Form login={handleLogin}/>
        </Route>
        <Route path="/dashboard" component={Dashboard}></Route>
        <Route path="*" component={ErrorPage}></Route>
      </Switch>
      </>
    </Router>
  );
}

export default App;

