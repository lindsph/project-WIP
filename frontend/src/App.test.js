import React from "react";
import 'mutationobserver-shim';
import { Router } from 'react-router-dom'
import { createBrowserHistory } from 'history'
import { render, fireEvent, container, getByText, act, getByRole } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect'
import App from "./App";

describe("<App />", () => {
    let history;
    beforeEach(() => {
        history = createBrowserHistory();
    });

    it('renders correctly', () => {
        render(<App />)
        expect(document.body).toMatchSnapshot();
    });

    // it('navigates to the sign in page when you click the "Sign in" link', () => {
    //     const { container, getByText } = render(<App />);
        
    //     // interact with the page
    //     act(() => {
    //         // find the link
    //         const signInLink = getByText("Sign in");

    //         fireEvent.click(signInLink);
    //     })

    //     expect(document.body.textContent).toContain('Email Address')
    // });

    it('page renders and can click on links to navigate', () => {
        const {container, getByText} = render(
            <Router history={history}>
                <App/>
            </Router>
        )

        // verify page content for expected route
        // MUSTDO: UPDATE THIS WHEN THE LANDING PAGE IS CREATED
        expect(container.innerHTML).toMatch(`I'm the landing`);

        fireEvent.click(getByText(/sign in/i))

        // check the content changed to the new page
        // MUSTDO: UPDATE THIS WHEN THE SIGN IN PAGE IS UPDATED?
        expect(container.innerHTML).toContain('Email Address')
    });

    it('routes to the 404 page for a bad URL', () => {
        history.push('/bad/route')

        const {getByRole} = render(
            <Router history={history}>
                <App />
            </Router>
        )

        expect(getByRole('heading')).toHaveTextContent('404 Not Found');
    })

});