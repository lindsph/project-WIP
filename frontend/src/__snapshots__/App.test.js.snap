// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<body>
  <div>
    <header>
      <nav>
        <a
          href="/sign_in"
        >
          Sign in
        </a>
      </nav>
    </header>
    <div>
      I'm the landing
    </div>
  </div>
</body>
`;
